### PR TITLE
feat: support for train transformers

### PR DESCRIPTION
**Pull request by s0nicboOm**
1. Introduce train transformers.
Note: train transformers are fitted first and then the preprocess transformers are fitted next.

2. DataClipper now just supports Sequence[float] . Avoiding just accepting a s ingle float value.

### CODE REVIEW

File: numalogic/config/_config.py [hunk 1]
```diff
@@ -94,8 +94,7 @@ class TrainerConf:
     retry_sec: int = 600  # 10 min
     batch_size: int = 64
     data_freq_sec: int = 60
-    # TODO: Support trainer based transform models
-    max_value_map: Optional[dict[str, float]] = None
+    transform: Optional[list[ModelInfo]] = None
```
Comment (ab93) [inline numalogic/config/_config.py:97]:
```suggestion
    transforms: Optional[list[ModelInfo]] = None
```

Comment (s0nicboOm) [reply numalogic/config/_config.py:97]: Thats a good catch!

File: tests/udfs/resources/_config.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ stream_confs:
     ml_pipelines:
       pipeline1:
         pipeline_id: "pipeline1"
-        metrics: [ "col1" , "col2" ]
+        metrics: [ "degraded" , "failed" ]
```
[the remark this answers is not in the document]
Comment (s0nicboOm) [reply tests/udfs/resources/_config.yaml:10]: Good catch!

File: numalogic/udfs/trainer/_base.py [hunk 3]
```diff
@@ -101,8 +103,13 @@ def compute(
             raise ConfigNotFoundError("Numalogic Trainer config not found!")
         dict_artifacts = {}
         trainer_cfg = numalogic_cfg.trainer
+        if trainer_transform:
+            input_ = trainer_transform.fit_transform(input_)
+            _LOGGER.info("Fit data using trainer transform")
```
Comment (ab93) [inline numalogic/udfs/trainer/_base.py:108]: Might be too many logs here

File: numalogic/udfs/trainer/_base.py [hunk 4]
```diff
@@ -101,8 +103,13 @@ def compute(
...
+
         if preproc_clf:
             input_ = preproc_clf.fit_transform(input_)
+            _LOGGER.info("Fit data using preproc transformer")
```
Comment (ab93) [inline numalogic/udfs/trainer/_base.py:112]: Doesn't really convey any meaningful info

File: numalogic/udfs/trainer/_base.py (hunk 3 above)
Comment (s0nicboOm) [reply numalogic/udfs/trainer/_base.py:108]: Oh this is just for testing the new changes. Let me remove it.

File: numalogic/udfs/trainer/_base.py (hunk 4 above)
Comment (s0nicboOm) [reply numalogic/udfs/trainer/_base.py:112]: same as above

**Pull request merged**